OrbitClientGgp/ClientGgp.cpp

```diff
@@ -79,9 +78,105 @@ bool ClientGgp::StopCapture() {
   return capture_client_->StopCapture();
 }
 
-void ClientGgp::InitCapture() {
+std::shared_ptr<Process> ClientGgp::GetOrbitProcessByPid(int32_t pid) {
+  // We retrieve the information of the process to later get the module corresponding to its binary
+  ErrorMessageOr<std::vector<ProcessInfo>> result_process_infos = process_client_->GetProcessList();
+  if (result_process_infos.has_error()) {
+    return nullptr;
+  }
+  const std::vector<ProcessInfo>& process_infos = result_process_infos.value();
+  LOG("List of processes:");
+  for (const ProcessInfo& info : process_infos) {
+    LOG("pid:%d, name:%s, path:%s, is64:%d", info.pid(), info.name(), info.full_path(),
+        info.is_64_bit());
+  }
+  auto process_it = find_if(process_infos.begin(), process_infos.end(),
+                            [&pid](const ProcessInfo& info) { return info.pid() == pid; });
+  if (process_it != process_infos.end()) {
+    LOG("Found process by pid, set target process");
+    std::shared_ptr<Process> process = std::make_shared<Process>();
+    process->SetID(process_it->pid());
+    process->SetName(process_it->name());
+    process->SetFullPath(process_it->full_path());
+    process->SetIs64Bit(process_it->is_64_bit());
+    LOG("Process info: pid:%d, name:%s, path:%s, is64:%d", process->GetId(), process->GetName(),
+        process->GetFullPath(), process->GetIs64Bit());
+    return process;
+  }
+  return nullptr;
+}
+
+bool ClientGgp::LoadModuleAndSymbols() {
```

Comment:
question - why do you need symbols?

Reply:
To be able to hook functions to a capture. I am working on that on a separate PR because this one was big enough already

Reply:
are these symbols limited to symbols from the main executable?

Reply:
at the moment, for simplicity, yes; if it is extended later I will update the code accordingly

Reply:
libggp and vulkan might make sense later. I agree that this should be added when needed.